### PR TITLE
add loading page

### PR DESCRIPTION
**Pull request by Trent1900**
### Descriptions
登陆页面
- Link to UI design (if possible):

### Provide screenshots of no warning, or error in the console.

- [ ] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [ ] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [ ] Check PR title is aligned with your ticket.
- [ ] Check your app will run without crashes.
- [ ] Check if there are hard-coded values.
- [ ] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [ ] Check you are using `camelCase && PascalCase` correctly.
- [ ] Remove unused comments.

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
+      Router.events.off("routeChangeError", end);
+    };
+  }, []);
   return (
     <WebAudioContextProvider>
       <MusicContextProvider>
         <GlobalLayout>
-          <Component {...pageProps} />
+          {isLoading ? (
+            <ImSpinner8 className="m-auto h-1/3 w-1/3 animate-spin text-primary" />
```
Comment (tatoMa) [inline src/pages/_app.tsx:33]: How about use h-1/2 w-1/2 -translate-x-1/2 -translate-y-1/2

Comment (tatoMa) [reply src/pages/_app.tsx:33]: To make the loading animation in the center of the main container.

Comment (Trent1900) [reply src/pages/_app.tsx:33]: that is great, will do

Comment (Trent1900) [reply src/pages/_app.tsx:33]: > How about use h-1/2 w-1/2 -translate-x-1/2 -translate-y-1/2

translate will be animated. see video, but I find that h-1/2 w-1/2 m-auto is not bad, see second video. 

https://github.com/taodemy/taotify/assets/63524682/cfc08a5e-8f6e-4ff3-b05b-7ce740fbdab6

https://github.com/taodemy/taotify/assets/63524682/618ce536-9942-40a5-966c-1349ed9293f3

Comment (Trent1900) [reply src/pages/_app.tsx:33]: I found h-2/5 w-2/5 looks very close to the effect of "middle". let's use this and move foreward :)

**Review by tatoMa — APPROVED** [review]
LGTM

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+    const end = () => {
+      setIsLoading(false);
+    };
+    Router.events.on("routeChangeStart", start);
+    Router.events.on("routeChangeComplete", end);
+    Router.events.on("routeChangeError", end);
+    return () => {
+      Router.events.off("routeChangeStart", start);
+      Router.events.off("routeChangeComplete", end);
+      Router.events.off("routeChangeError", end);
```
Comment (tatoMa) [inline src/pages/_app.tsx:25]: These lines can be a custom hook If you like to extract them.

**Pull request merged**